src/lambda/local/configureLocalLambda.ts

```diff
@@ -0,0 +1,245 @@
+/*!
+ * Copyright 2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+'use strict'
+
+import { parse } from 'jsonc-parser'
+import * as os from 'os'
+import * as path from 'path'
+import * as vscode from 'vscode'
+import { accessAsync, mkdirAsync, writeFileAsync } from '../../shared/filesystem'
+import { DefaultSettingsConfiguration } from '../../shared/settingsConfiguration'
+
+export interface HandlerConfig {
+    event: any
+}
+
+export interface HandlersConfig {
+    handlers: {
+        [ handler: string ]: HandlerConfig | undefined
+    }
+}
+
+export async function configureLocalLambda(sourceUri: vscode.Uri, handler: string): Promise<void> {
+    // Handler will be the fully-qualified name, so we also allow '.' despite it being forbidden in handler names.
+    if (/[^\w\-\.]/.test(handler)) {
+        throw new Error(
+            `Invalid handler name: '${handler}'. ` +
+            'Handler names can contain only letters, numbers, hyphens, and underscores.'
+        )
+    }
+
+    const workspaceFolder: vscode.WorkspaceFolder | undefined = vscode.workspace.getWorkspaceFolder(sourceUri)
+    if (!workspaceFolder) {
+        console.error(`Source file ${sourceUri} is external to the current workspace.`)
+
+        return
+    }
+
+    // Preserve the scheme, etc of the workspace uri.
+    const uri = workspaceFolder.uri.with({
+        path: path.join(workspaceFolder.uri.fsPath, '.aws', 'handlers.json')
+    })
+
+    try {
```

Comment:
Consider making `ensureHandlerJsonExists` method to encapsulate all create-if-not-exists code

Reply:
✔️